### PR TITLE
connections tracking

### PR DESCRIPTION
**Pull request by o-mehlii**
https://trello.com/c/H4XWpXn9/340-how-many-calendars-are-connected-metric

### CODE REVIEW

**Review by a-nahaitsev — APPROVED** [review]

File: renderer/src/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -193,6 +194,11 @@ export function Calendar({
     });
   }, [selectedDate]);
 
+  useEffect(() => {
+    if (timetrackerUserInfo) {
+      trackConnections("timetrackerWeb");
```
Comment (okhmelevskyi) [inline renderer/src/components/Calendar/Calendar.tsx:199]: Create a constant for trackConnections args.
An example `const TRACK_CONNECTIONS = { TIMETRACKER_WEB: 'timetrackerWeb' ... };`

File: renderer/src/helpers/utils/google.ts [hunk 2]
```diff
@@ -37,6 +38,8 @@ export const loadGoogleEventsFromAllUsers = async () => {
 
   if (!storedUsers.length) return [];
 
+  trackConnections("google-calendar");
```
Comment (okhmelevskyi) [inline renderer/src/helpers/utils/google.ts:41]: The same here.

File: renderer/src/helpers/utils/jira.ts [hunk 3]
```diff
@@ -76,6 +77,8 @@ export const getJiraResources = async () => {
 
     if (!storedUsers.length) return [];
 
+    trackConnections("jira");
```
Comment (okhmelevskyi) [inline renderer/src/helpers/utils/jira.ts:80]: The same here.

File: renderer/src/helpers/utils/office365.ts [hunk 4]
```diff
@@ -59,6 +59,8 @@ export const getOffice365Events = async () => {
 
   if (!storedUsers.length) return [];
 
+  trackConnections("office365-calendar");
```
Comment (okhmelevskyi) [inline renderer/src/helpers/utils/office365.ts:62]: The same here.

File: renderer/src/helpers/utils/trello.ts [hunk 5]
```diff
@@ -15,6 +16,8 @@ export const getAllTrelloCardsFromApi = async () => {
 
   if (!user) return [[], []];
 
+  trackConnections("trello");
```
Comment (okhmelevskyi) [inline renderer/src/helpers/utils/trello.ts:19]: The same here.

File: renderer/src/helpers/utils/utils.ts [hunk 6]
```diff
@@ -137,3 +138,24 @@ export function extractTokenFromString(inputString: string) {
 
   return "";
 }
+
+export const trackConnections = (connectedName: string) => {
```
Comment (okhmelevskyi) [inline renderer/src/helpers/utils/utils.ts:142]: Could you please cover this new function with unit tests? And check all cases.

**Review by okhmelevskyi — CHANGES_REQUESTED** [review]

**Review by okhmelevskyi — APPROVED** [review]

**Pull request merged**